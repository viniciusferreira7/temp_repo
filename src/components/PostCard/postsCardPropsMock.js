export const postsCardPropsMock = {
  posts: {
    title: 'title 1',
    body: 'body 1',
    cover: 'img/img.png',
  },
};
