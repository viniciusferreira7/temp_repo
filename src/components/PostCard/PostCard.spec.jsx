import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postsCardPropsMock } from './postsCardPropsMock';

const props = postsCardPropsMock;

describe('<PostsCard/>', () => {
  it('should render the PostsCard correctly', () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name: props.posts.title })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: props.posts.title })).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: props.posts.title })).toBeInTheDocument();
    expect(screen.getByText('body 1')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container).toMatchSnapshot();
  });
});
