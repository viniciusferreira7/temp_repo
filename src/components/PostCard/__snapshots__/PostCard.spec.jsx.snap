// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsCard/> should match snapshot 1`] = `
<div>
  <div
    class="post"
  >
    <img
      alt="title 1"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h3>
        title 1
      </h3>
      <p>
        body 1
      </p>
    </div>
  </div>
</div>
`;
